tests: Add xg29_rb4412a in various test cases (samples/tests)

Martinhoff-maker opened the pull request:
The goal of this PR is to run CI on more test case and sample test case for Silabs xg29_rb4412a platform.

asmellby commented on tests/drivers/uart/uart_async_api/boards/xg29_rb4412a.overlay:
```diff
@@ -0,0 +1,67 @@
+ */
+
+/*
+ * Need to connect PB0 and PB1 of the Expension Pin header
```
Comment says PB0 and PB1, but pinctrl assigns PB2 and PB3.
These pins are also different from the GPIO loopback test above. It would be useful if the same loopback connection was used for as many tests as possible, and that different loopback pairs never conflict.

asmellby commented on samples/drivers/adc/adc_dt/boards/xg29_rb4412a.overlay:
```diff
@@ -0,0 +1,50 @@
+		zephyr,vref-mv = <3300>;
+		zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
+		zephyr,resolution = <12>;
+		zephyr,input-positive = <IADC_INPUT_PB1>;
```
Should this sample use PD3 instead? It is connected to the joystick, which allows for multiple analog measurements, as opposed to PB1, which is connected to a button that only has VDD and GND states. The connection to the joystick isn't present on xg24_rb4187c (0 ohm resistor not mounted), which is why it uses a button.

Martinhoff-maker committed:
samples: Add silasb xg29_rb4412a to some sample.yaml with overlay

Enable to test samples on silabs xg29_rb4412a board.

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>

Martinhoff-maker committed:
tests: drivers: add overlay/testcase for silabs xg29_rb4412a board

Add some overlay files for the silabs xg29_rb4412a board to enable tests
on the board. Also add the platform to some testcase.yaml files.

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>

Martinhoff-maker committed:
boards: silabs: Update xg29_rb4412a.yaml file

Add new tag to xg29_rb4412a.yaml file and remove pm ignore tag in order
to enable more tests for the xg29_rb4412a board.

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>

jerome-pouiller reviewed (approved):
Great!
